src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java

```diff
@@ -600,29 +604,56 @@ public Response showSettings(@Context HttpServletRequest request) throws IOExcep
 	@POST
 	@Path("/branches/delete")
 	@Transactional
-	public Response deleteBehindBranch(@Context HttpServletRequest request,
-									   @FormParam("branchDeleteName") String branchDeleteName,
-									   @QueryParam("fatal") String fatal)
+	public Response deleteBranch(@Context HttpServletRequest request,
+                                 @FormParam("branchDeleteName") String branchDeleteName,
+                                 @FormParam("branchDeleteNameConfirmation")
+                                             String branchDeleteNameConfirmation,
+                                 @QueryParam("fatal") String fatal)
 			throws IOException, ApiError {
 
 		RepositoryEntity repositoryEntity = getRepositoryEntity();
 		RepositoryApi repositoryApi = repositoriesApi.getRepository(repositoryEntity.getRepositoryName());
 
         List<Locale> locales = Collections.list(request.getLocales());
 
-        Map<String, Object> parameters = getBaseParameters();
+        Map<String, Object> parameters;
```

Comment:
Just use `getBaseParameters()` here, and add the `BranchModel` for the current branch under a variable. This gives you access in Freemarker to both the simple name and the actual name.